### ISSUE TITLE
Dynamic Attributes

### ISSUE THREAD

**Issue by ShMcK** [angular/material#1478]
I recently attempted to create [`angular-formly` templates for `angular-material`](https://github.com/formly-js/angular-formly-templates-material). This project was unsuccessful as `angular-material` is doing some strange stuff with compiled attributes. 

I'm not sure what is happening to attributes, but I don't think it is a best practice. Perhaps they are being parsed, or `class` is being treated as its own directive rather than the `html` default. 

For example, styles, which rely on `class`, cannot be added.

```html
<md-checkbox ng-model="model[options.key]"
             aria-label="{{::options.templateOptions.label}}"
             ng-class="::options.templateOptions.className"> 
	{{::options.templateOptions.label}}
</md-checkbox>
```

When compiled, `ng-class` is ignored in this template, class is unchanged. In other words, templates can't change the color or style. This is true for all `angular-material` form directives except `md-slider`.

When dynamically adding other `ng-directives`, they are added to the attribute as `ngdirective`, without 
the `-`. For example, adding `ng-required` in the compile stage results in:

```html
<md-checkbox ngrequired=true></md-checkbox>
```

Wha??

It's as if `angular-material` has removed all the dynamic functionality that makes `angular` great, and replaced it with their own static directives. Users should be able to add any directives they'd like to a form directive.

I've also made similar [templates for `LumX`](https://github.com/formly-js/angular-formly-templates-lumx) which work great. You can view them [online](https://formly-lumx.herokuapp.com/). I would love to make templates available for `angular-material`, however, this does not seem currently possible. 

In conclusion, please make `angular-material` more in line with `html` standards, and remove attribute parsing so users won't run into framework specific road blocks down the road. I think `angular-material` has a lot of potential, and I hope to see more good stuff.

**Comment by ThomasBurleson** [comment]
Please post this in the [Angular Material Forum](https://groups.google.com/forum/#!forum/ngmaterial)
